services/121-service/src/notifications/whatsapp/whatsapp.service.ts

```diff
@@ -103,20 +101,42 @@ export class WhatsappService {
     const language = preferedLanguage
       ? preferedLanguage
       : this.fallbackLanguage;
-    const instance = await this.instanceRepository.findOne();
-    const senderDisplay = instance.displayName[preferedLanguage];
-    const genericReplyMessage = whatsappGenericMessage[language]
-      .split('{{1}}')
-      .join(senderDisplay);
+    const registration = await this.registrationRepository.findOne(
+      registrationId,
+      {
+        relations: ['program'],
+      },
+    );
+    const whatsappGenericMesage = this.getGenericNotificationText(
+      language,
+      registration.program,
+    );
     this.sendWhatsapp(
-      genericReplyMessage,
+      whatsappGenericMesage,
       recipientPhoneNr,
       messageType,
       mediaUrl,
       registrationId,
     );
   }
 
+  public getGenericNotificationText(
+    language: string,
+    program: ProgramEntity,
+  ): string {
+    const key = 'whatsappGenericMesage';
+    const fallbackNotifications = program.notifications[this.fallbackLanguage];
+    let notifications = fallbackNotifications;
+
+    if (program.notifications[language]) {
+      notifications = program.notifications[language];
+    }
+    if (notifications[key]) {
+      return notifications[key];
+    }
+    return fallbackNotifications[key] ? fallbackNotifications[key] : '';
```

Comment:
@RubenGeo FYI I believe you can write this shorter as 'return fallbackNotifications[key] || ''; (no need to change now)